src/Service/Update.php

```diff
@@ -0,0 +1,145 @@
+use GuzzleHttp\Exception\GuzzleException;
+use Outlandish\Wpackagist\Package\AbstractPackage;
+use Outlandish\Wpackagist\Package\Plugin;
+use Outlandish\Wpackagist\Package\Theme;
+use Rarst\Guzzle\WporgClient;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class Update
+{
+    /** @var Connection */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function update(OutputInterface $output, ?string $name = null)
+    {
+        $updateStmt = $this->connection->prepare(
+            'UPDATE packages SET
+            last_fetched = datetime("now"), versions = :json, is_active = 1, display_name = :display_name
+            WHERE class_name = :class_name AND name = :name'
+        );
+        $deactivateStmt = $this->connection->prepare('UPDATE packages SET last_fetched = datetime("now"), is_active = 0 WHERE class_name = :class_name AND name = :name');
+
+        if ($name) {
+            $query = $this->connection->prepare('
+                SELECT * FROM packages
+                WHERE name = :name
+            ');
+            $query->bindValue("name", $name);
+        } else {
+            $query = $this->connection->prepare('
+                SELECT * FROM packages
+                WHERE last_fetched IS NULL
+                OR last_fetched < datetime(last_committed, "+2 hours")
+                OR (is_active = 0 AND last_committed > date("now", "-90 days") AND last_fetched < datetime("now", "-7 days"))
+            ');
+        }
+        // get packages that have never been fetched or have been updated since last being fetched
+        // or that are inactive but have been updated in the past 90 days and haven't been fetched in the past 7 days
+        $query->execute();
+        $packages = $query->fetchAll(\PDO::FETCH_CLASS | \PDO::FETCH_CLASSTYPE);
+
+        $count = count($packages);
+        $versionParser = new VersionParser();
+
+
+        $wporgClient = WporgClient::getClient();
+
+        foreach ($packages as $index => $package) {
+
+            $percent = $index / $count * 100;
+
+            if ($package instanceof Plugin) {
+                try {
+                    $info = $wporgClient->getPlugin($package->getName(), ['versions']);
+                } catch (GuzzleException $exception) {
+                    $output->writeln("Skipped plugin '{$package->getName()}' due to error: '{$exception->getMessage()}'");
+                }
+            } else {
+                try {
+                    $info = $wporgClient->getTheme($package->getName(), ['versions']);
+                } catch (GuzzleException $exception) {
+                    $output->writeln("Skipped theme '{$package->getName()}' due to error: '{$exception->getMessage()}'");
+                }
+            }
+
+            $output->writeln(sprintf("<info>%04.1f%%</info> Fetched %s", $percent, $package->getName()));
+
+            if (empty($info)) {
```

Comment:
This can only ever be empty if the first x packages are empty - it is never set to something falsey (getPlugin/getTheme should always return an array or throw an exception)
This is causing plugins to not be deactivated properly, and have the wrong information assigned to them

Reply:
@rasmuswinter It is left unset (`null` or `undefined` – I haven't done a full audit of all existing code) in (at least some) exception cases, because `GuzzleException`s are caught, logged and not re-thrown above.

This logic is also not new in this PR, I just moved it from the `Command` so I could get rid of the web-calls-CLI hack.

Reply:
But `$info` is not reset in each loop. I've tried running it locally, and can see that there are inconsistencies with the package information with all inactive plugins. I've got a fix locally, so can push that to this branch, if that's ok?

Reply:
Ah got you @rasmuswinter, yep that's good with me, thanks.